docs/proposals/remediation/cluster-status-based-remedy.md

```diff
@@ -0,0 +1,383 @@
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+The detection accuracy of the domain name resolution function in the cluster is not 100% correct. When users find that the domain name resolution function in the cluster is unavailable through other tools or relying on operation and maintenance experience, they can proactively report it and synchronize the information to the condition of the Cluster object.
+
+## Design Details
+
+### Cluster status management
+
+We do cluster status management by extending the ConditionType in the Cluster object. A new `ServiceDomainNameResolutionReady` ConditionType is added to represent the domain name resolution function within the cluster.
+
+
+
+Let us show the cluster status through Yaml example.
+
+```yaml
+apiVersion: "cluster.karmada.io/v1alpha1"
+kind: "Cluster"
+metadata:
+name: "member1"   
+spec:
+apiEndpoint: "https://192.10.0.1:6443"
+syncMode: Push
+status:
+conditions:
+- lastTransitionTime: "2024-01-25T04:05:57Z"
+  message: "cluster is healthy and ready to accept workloads"
+  status: "True"
+  type: Ready
+- lastTransitionTime: "2024-01-25T04:05:57Z"
+  message: "coredns function is normal "
+  status: "True"
+  type: ServiceDomainNameResolutionReady
+```
+
+The above example records that the Service domain name resolution function in cluster member1 is normal.
+
+When an exception occurs in the Service domain name resolution function in cluster member1, the cluster object is displayed as follows:
+
+```yaml
+apiVersion: "cluster.karmada.io/v1alpha1"
+kind: "Cluster"
+metadata:
+name: "member1"   
+spec:
+apiEndpoint: "https://192.10.0.1:6443"
+syncMode: Push
+status:
+conditions:
+- lastTransitionTime: "2024-01-25T04:05:57Z"
+  message: "cluster is healthy and ready to accept workloads"
+  status: "True"
+  type: Ready
+- lastTransitionTime: "2024-01-25T04:05:57Z"
+  message: "coredns function is not normal "
+  status: "False"
+  type: ServiceDomainNameResolutionReady
+ ```
+
+### Cluster status detection
+
+The cluster status detection mechanism can detect active components directly without using the kube-apiserver of member clusters to avoid intermediate component failures affecting overall fault detection and response. In terms of the specific live detection effect, the active cluster group function is Availability, not just component survival.
+
+#### Component build and deployment
+
+Deploy the agent component in the form of DaemonSet in the Karmada member cluster to automatically detect cluster status changes in the cluster and report the detection results to the Karmada control plane. 
```

Comment:
> Deploy the agent component in the form of DaemonSet in the Karmada member cluster to automatically detect cluster status changes in the cluster and report the detection results to the Karmada control plane. 


Why not adopt some commonly used tools in the K8s ecosystem?  

e.g. https://github.com/kubernetes/node-problem-detector

Reply:
A separate design solution is required to perform in-depth analysis of the cluster status detection feature. In the current proposal, only a sample is provided for users to perform end-to-end tests.